### DISCUSSION TITLE
Using Prisma with Databricks / with other jdbc databases

### DISCUSSION THREAD

**Discussion by J-Bradlee** [prisma/prisma#25239] [Q&A]
### Question

Hi,

I was wondering if there was a way to use Prisma with Databricks/Spark clusters. Databricks has a [JDBC driver](https://docs.databricks.com/en/integrations/jdbc/index.html) and a [Node js driver](https://docs.databricks.com/en/dev-tools/nodejs-sql-driver.html) is there a way to easily integrate this with Prisma? 

What are some of the overall steps to use Prisma with a non-supported relational database types? From reading the docs, perhaps something to do with the query engine aspect of Prisma?

Generally, just trying to understand the feasibility of using Prisma with my current Databricks stack.

**Comment by ludralph** [comment]
Hi @J-Bradlee 👋 

Thank you for raising this question.

Could you clarify the specific problem and use case you want to address? How do you envision Prisma integration fitting into your current Databricks stack, and what functionality are you expecting from this support?

**Comment by J-Bradlee** [comment]
Thanks for replying.

**My current problem:**
I have most of my data stored as very large delta tables on databricks which are fundamentally stored as many parquet files, and are accessed via a spark cluster to run queries on using sparkSQL. These delta tables have ACID guarantees and effectively work very similarly to mySQL. I access these tables for a web app which serves that data to my end users. Changes in schema to these tables can cause a headache to fix/debug on the type script web app.

**How I envision Prisma integratting into my stack:**
I would like to have the same end-to-end type safety features you get from the prisma client with something like mySQL but with a [Delta table](https://delta.io/) as a source (or really any sort of big data sql abstraction format, for example: [apache iceberg](https://iceberg.apache.org/).  I want to use the auto-generated prisma client to help manage any schema changes I have to the underlying tables for my web app.

**Functionality I expect from the support:**
I would ideally like all the features of the Prisma Client (as much as possible, although I understand there may be some fundamental differences, such as transactions which are different with delta tables that normal relational database management systems). 

I think prisma makes life easier for server-side apps that talks to a database, I'd like to extend this to some very large data sources which aren't supported by current  relational database management systems.

**Reply by ludralph** [reply]
Hi @J-Bradlee 

Thank you for your detailed response and for explaining your use case further. Unfortunately, Prisma currently does not support Databricks, Delta Lake, or any other big data SQL abstraction formats like Apache Iceberg. Prisma only supports the relational databases listed here: [Prisma Supported Databases](https://www.prisma.io/docs/orm/reference/supported-databases). To support Databricks, this would require the creation of a custom connector. I have shared your use case with our engineering team for consideration.

**Reply by J-Bradlee** [reply]
I appreciate the response cheers!

**Reply by The-Lone-Druid** [reply]
It would be really helpful if we had such integration, which would help build more complex and unique applications.
